### ISSUE TITLE
no-return-assign: except-parens does not behave correctly on arrow syntax

### ISSUE THREAD

**Issue by CapacitorSet** [eslint/eslint#7285]
**Tell us about your environment**

* **ESLint Version:** v3.7.0
* **Node Version:** v6.7.0
* **npm Version:** v3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```json
module.exports = {
    "extends": "google",
    "rules": {
        "camelcase": 1,
        "curly": 0,
        "eol-last": 0,
        "indent": ["error", "tab", {"SwitchCase": 1}],
        "max-len": 0,
        "new-cap": 1,
        "no-else-return": 1,
        "no-extend-native": 0,
        "no-loop-func": 0,
        "no-unused-vars": 1,
        "require-jsdoc": 0
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
function Dictionary() {
	this.dictionary = {};
	this.add = (key, value) => (this[key] = value);
	this.item = key => this[key];
}
```

**What did you expect to happen?**

No error nor warning, as per [documentation](http://eslint.org/docs/rules/no-return-assign#except-parens): `no-return-assign` throws an error unless `"except-parens"` is enabled (like the default) and the assignment is enclosed in parentheses.

**What actually happened? Please include the actual, raw output from ESLint.**

`eslint` prints this: `3:13  error  Arrow function should not return assignment  no-return-assign`

**Comment by not-an-aardvark** [comment]
Thanks for the issue. However, I was unable to reproduce it in the demo:

![](https://i.gyazo.com/0016d3d4f9721841addf0edbb60416be.png)

I don't see `no-return-assign` in your config. Is it possible that the option is being set to `no-return-assign: [2, "always"]`?

(You can view the full config for a file by using `eslint --print-config theFile.js`.)

**Comment by CapacitorSet** [comment]
Oh, yeah, that's why - thank you. I can't seem to find it in `eslint-config-google`, but if I replace `"extends": "google"` with `"extends": "standard"` it works without issues.
